torch/csrc/distributed/autograd/context/container.cpp

```diff
@@ -41,8 +41,16 @@ DistAutogradContainer& DistAutogradContainer::init(int64_t worker_id) {
 
   auto& container = getInstanceInternal();
   TORCH_CHECK(
-      !container.initialized_,
-      "Container is already initialized! Cannot initialize it twice!");
+      !container.initialized_ || (worker_id == container.worker_id_),
```

Comment:
Does this mean if we shut down RPC, and then re-init, we can't change the passed in rank? I don't think there's a reason to do so, but just confirming.

Reply:
Yes, that is correct. I think it is probably safer to not allow rank changes and its easier to also just reuse the same container.